File: woodroid/src/main/java/me/gilo/woodroid/data/api/ShippingZoneAPI.java
package me.gilo.woodroid.data.api;


import me.gilo.woodroid.models.Coupon;
import me.gilo.woodroid.models.ShippingZone;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public interface ShippingZoneAPI {

    @Headers("Content-Type: application/json")
    @POST("shipping/zones")
    Call<ShippingZone> create(@Body ShippingZone body);

    @GET("shipping/zones/{id}")
    Call<ShippingZone> view(@Path("id") int id);

    @GET("shipping/zones")
    Call<List<ShippingZone>> list();

    @Headers("Content-Type: application/json")
    @PUT("shipping/zones/{id}")
    Call<ShippingZone> update(@Path("id") int id, @Body ShippingZone body);

    @DELETE("shipping/zones/{id}")
    Call<ShippingZone> delete(@Path("id") int id);

    @DELETE("shipping/zones/{id}")
    Call<ShippingZone> delete(@Path("id") int id, @Query("force") boolean force);

    @POST("shipping/zones/batch")
    Call<String> batch(@Body ShippingZone body);

    @GET("coupons")
    Call<ArrayList<ShippingZone>> filter(@QueryMap Map<String, String> filter);

}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/ShippingZoneAPI.kt
package me.gilo.woodroid.data.api


import me.gilo.woodroid.models.ShippingZone
import retrofit2.Call
import retrofit2.http.*

import java.util.ArrayList

interface ShippingZoneAPI {

    @Headers("Content-Type: application/json")
    @POST("shipping/zones")
    fun create(@Body body: ShippingZone): Call<ShippingZone>

    @GET("shipping/zones/{id}")
    fun view(@Path("id") id: Int): Call<ShippingZone>

    @GET("shipping/zones")
    fun list(): Call<List<ShippingZone>>

    @Headers("Content-Type: application/json")
    @PUT("shipping/zones/{id}")
    fun update(@Path("id") id: Int, @Body body: ShippingZone): Call<ShippingZone>

    @DELETE("shipping/zones/{id}")
    fun delete(@Path("id") id: Int): Call<ShippingZone>

    @DELETE("shipping/zones/{id}")
    fun delete(@Path("id") id: Int, @Query("force") force: Boolean): Call<ShippingZone>

    @POST("shipping/zones/batch")
    fun batch(@Body body: ShippingZone): Call<String>

    @GET("coupons")
    fun filter(@QueryMap filter: Map<String, String>): Call<ArrayList<ShippingZone>>

}